src/Common/src/System/Net/CaseInsensitiveAscii.cs

```diff
@@ -85,46 +85,39 @@ public int Compare(object firstObject, object secondObject)
             return result;
         }
 
-        // ASCII string case insensitive hash function
-        private int FastGetHashCode(string myString)
-        {
-            int myHashCode = myString.Length;
-            if (myHashCode != 0)
-            {
-                myHashCode ^= AsciiToLower[(byte)myString[0]] << 24 ^ AsciiToLower[(byte)myString[myHashCode - 1]] << 16;
-            }
-            return myHashCode;
-        }
-
         // ASCII string case insensitive comparer
         public new bool Equals(object firstObject, object secondObject)
         {
             string firstString = firstObject as string;
             string secondString = secondObject as string;
-            if (firstString == null)
+            if ((object)firstString == (object)secondString)
```

Comment:
This could be slightly faster again if we did `if (firstObject == secondObject)` before the casts, and I'm 99.9% sure this would never be called with non-strings and hence have a change in behaviour because of it, but I'm putting it here because of that 0.1% of doubt.